### ISSUE TITLE
HF calculation for large basis size returns without error message

### ISSUE THREAD

**Issue by y-yao** [pyscf/pyscf#774]
When I run HF calculations on the same system (ScO), 2z and 3z can finish but 4z returns before starting the HF calculation with no error message. In particular, at verbosity level 6 the end of the output I get is copied below along with an empty tmp* file in the directory.
The version I'm using is v1.5.4 from Oct 2020. I've also done a check with a much older version from 2018: there, 4z is able to finish.
Is this a memory-related issue?
Thanks.

******** <class 'pyscf.scf.hf_symm.SymAdaptedROHF'> Newton solver flags ********
SCF tol = 1e-06
conv_tol_grad = None
max. SCF cycles = 100
direct_scf = True
direct_scf_tol = 1e-13
chkfile to save SCF result = /data/yuanyao/monoxides_dft/ScO/4z/tmp03a7gia1
max_cycle_inner = 12
max_stepsize = 0.05
ah_start_tol = 1e+09
ah_level_shift = 0
ah_conv_tol = 1e-12
ah_lindep = 1e-14
ah_start_cycle = 1
ah_max_cycle = 40
ah_grad_trust_region = 2.5
kf_interval = 4
kf_trust_region = 5
canonicalization = True
max_memory 4000 MB (current use 70 MB)
Set conv_tol_grad to 0.001
Initial guess density matrix is not given. Generating initial guess from minao
Nelec from initial guess = (8.5, 8.5)

**Comment by fishjojo** [comment]
seems like the same problem as in #742. what if you downgrade libcint to older version?

**Comment by y-yao** [comment]
@fishjojo 
Hi Xing, the original version I was using had libcint v4.0.5 in CMakeLists.txt. I changed it to v3.0.20 as was reported in #742 and it still returned at the same point. I saw that the latest version on github had v4.0.6 so I made a git pull but it made no difference. 
Compared to #742, one difference I have is that I don't see any seg fault error. The job just ends there.

**Comment by fishjojo** [comment]
Can you share the input so that I can test it. Thanks.

**Comment by y-yao** [comment]
Here's a more self-contained example that shows the same behavior (works for 2z 3z, but not 4z).

```python
from pyscf import gto
from pyscf import scf

b = 1.68
mol = gto.Mole()
mol.verbose = 5
mol.atom = [
    ['Cr',(  0.000000,  0.000000, -b/2)],
    ['Cr',(  0.000000,  0.000000,  b/2)],
]
mol.basis = 'ccpvtzdk'
mol.symmetry = 1
mol.build()
mf = scf.RHF(mol)
mf.level_shift = .4
mf.max_cycle = 100
mf.conv_tol = 1e-9

mf.scf()  # do HF
```

The end of the output for 4z is:
******** <class 'pyscf.scf.hf_symm.SymAdaptedRHF'> ********
method = SymAdaptedRHF-RHF
initial guess = minao
damping factor = 0
level_shift factor = 0.4
DIIS = <class 'pyscf.scf.diis.CDIIS'>
diis_start_cycle = 1
diis_space = 8
SCF conv_tol = 1e-09
SCF conv_tol_grad = None
SCF max_cycles = 100
direct_scf = True
direct_scf_tol = 1e-13
chkfile to save SCF result = /data/yuanyao/Cr2/4z/tmptqt884rb
max_memory 4000 MB (current use 65 MB)
Freeze 0 electrons in irreps []
    48 free electrons in irreps A1g E1gx E1gy A1u E1uy E1ux E2gx E2gy E3gx E3gy E2uy E2ux E3uy E3ux E4gx E4gy E5gx E5gy E4uy E4ux E5uy E5ux
Set gradient conv threshold to 3.16228e-05
Nelec from initial guess = 47.999894612713966

**Comment by sunqm** [comment]
This is a bug in libcint. The latest pypi release should work. If you compile
things from source code, you need to install libcint-4.0.7 manually.

On Thu, Dec 03, 2020 at 08:48:43PM -0800, Yuan Yao wrote:
> Here's a more self-contained example that shows the same behavior (works for 2z 3z, but not 4z).
> 
> ```python
> from pyscf import gto
> from pyscf import scf
> 
> b = 1.68
> mol = gto.Mole()
> mol.verbose = 5
> mol.atom = [
>     ['Cr',(  0.000000,  0.000000, -b/2)],
>     ['Cr',(  0.000000,  0.000000,  b/2)],
> ]
> mol.basis = 'ccpvtzdk'
> mol.symmetry = 1
> mol.build()
> mf = scf.RHF(mol)
> mf.level_shift = .4
> mf.max_cycle = 100
> mf.conv_tol = 1e-9
> 
> mf.scf()  # do HF
> ```
> 
> The end of the output for 4z is:
> ******** <class 'pyscf.scf.hf_symm.SymAdaptedRHF'> ********
> method = SymAdaptedRHF-RHF
> initial guess = minao
> damping factor = 0
> level_shift factor = 0.4
> DIIS = <class 'pyscf.scf.diis.CDIIS'>
> diis_start_cycle = 1
> diis_space = 8
> SCF conv_tol = 1e-09
> SCF conv_tol_grad = None
> SCF max_cycles = 100
> direct_scf = True
> direct_scf_tol = 1e-13
> chkfile to save SCF result = /data/yuanyao/Cr2/4z/tmptqt884rb
> max_memory 4000 MB (current use 65 MB)
> Freeze 0 electrons in irreps []
>     48 free electrons in irreps A1g E1gx E1gy A1u E1uy E1ux E2gx E2gy E3gx E3gy E2uy E2ux E3uy E3ux E4gx E4gy E5gx E5gy E4uy E4ux E5uy E5ux
> Set gradient conv threshold to 3.16228e-05
> Nelec from initial guess = 47.999894612713966
> 
> -- 
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/pyscf/pyscf/issues/774#issuecomment-738562495

--

**Comment by y-yao** [comment]
@sunqm 
Hi Qiming, I tried libcint-4.0.7 and got the same behavior.
To be sure I'm doing what you're suggesting, I did a git clone of the master branch of pyscf, changed L92 of CMakelists.txt to `GIT_TAG v4.0.7` and compiled again. Or is something else I need to do manually?

**Comment by sunqm** [comment]
Are there other pyscf libraries mixed in your runtime environment? This is a known
issue which should be fixed in cint-4.0.7 release.

On Fri, Dec 04, 2020 at 08:13:06AM -0800, Yuan Yao wrote:
> @sunqm 
> Hi Qiming, I tried libcint-4.0.7 and got the same behavior.
> To be sure I'm doing what you're suggesting, I did a git clone of the master branch of pyscf, changed L92 of CMakelists.txt to `GIT_TAG v4.0.7` and compiled again. Or is something else I need to do manually?
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/pyscf/pyscf/issues/774#issuecomment-738870731

--

**Comment by y-yao** [comment]
It turned out that simply deleting lib/build/ and recompiling didn't completely get rid of the previous compilation. (Are there files stored elsewhere in the directory?)
So I cloned a fresh copy and recompiled and things are working now. 
Thanks Qiming and Xing!
